### PR TITLE
fix: suppressed some of the warnings

### PR DESCRIPTION
**Pull request by khushal-banks**
- modified type-specifiers for ptrdiff_t in fprintf to %td
- modified return type of int_koliseo_version() to int
- added const as type specifier to some of the pointers which are not modified within a function

> [!NOTE]
> Tested on `linux` with `gcc` 13.2.1

### CODE REVIEW

**Review by jgabaut — CHANGES_REQUESTED** [review]
Please clarify my doubts, this look like a nice fix PR.

File: src/koliseo.c
Comment (khushal-banks) [reply src/koliseo.c:763]: Right side ...  PTRDIFF_MAX / size is a fraction. long size fits. So, left that part unchanged. Seems correct.
Before you merge this; i want to know if `workflow` also checks for windows build?

File: src/koliseo.c
Comment (khushal-banks) [reply src/koliseo.c:2714]: > Is it really only responsibility of caller to ensure the freed temp pointer is not usable anymore?
```                                                                                                                                     
// current requirement:                                                                                                                 
kls_temp_end(tmp_kls);                                                                                                                      
tmp_kls = NULL;                                                                                                                             
```

> Should we do:
> `*tmp_kls = (Koliseo_Temp){0};`

Are you talking about clearing tmp_kls from within the `kls_temp_end` function ?
If yes, i think we would need following changes:
```
// syntax change
void kls_temp_end(Koliseo_Temp **address_tmp_kls);
void kls_temp_end(Koliseo_Temp **address_tmp_kls) {
Koliseo_Temp *tmp_kls = *address_tmp_kls;
....
....
*address_tmp_kls = NULL; // this will clear tmp_kls; and caller wouldn't have to do it
}
```

```
// Caller would then simply call
kls_temp_end(&tmp_kls);  
```

> This one I'mis something I'm not sure, better to leave it out of this PR.
> Is this line fine as it is?

I was not sure how you want to tackle this. I suggested best way i know.
I left the comment to indicate this is a known issue/requirement for this function which is left as it is for this PR

**Review by khushal-banks — COMMENTED** [review]
Added some comments.

File: src/koliseo.c
Comment (jgabaut) [reply src/koliseo.c:763]: It does not, I will have to try the build myself.
I'll update this PR soon, give me a day to find some time.

**Comment by jgabaut** [comment]
Alright, just give me some time to test the build myself and this should go in smoothly.

**Review by jgabaut — APPROVED** [review]

**Pull request merged**
